lib/Sema/TypeCheckAccess.cpp

```diff
@@ -1883,6 +1884,11 @@ bool isFragileClangNode(const ClangNode &node) {
     return isFragileClangType(pd->getType());
   if (auto *typedefDecl = dyn_cast<clang::TypedefNameDecl>(decl))
     return isFragileClangType(typedefDecl->getUnderlyingType());
+  if (auto *rd = dyn_cast<clang::RecordDecl>(decl)) {
+    if (!isa<clang::CXXRecordDecl>(rd))
+      return false;
+    return !rd->getDeclContext()->isExternCContext();
```

Comment:
Would we want to consider a struct that is declared within another struct declared within an `extern "C"` block as resilient?

Reply:
I did some experiment, and in C, we cannot really do the following:
```
struct Outer {
  struct Inner {
  };
};

void f(Outer::Inner x); // This does not compile in C.
```

While the above would work in C++, it does not work in C. So, I think this approach should work OK for C headers.

That being said, this would work for C++:
```
extern "C" {
struct Outer {
  struct Inner {
  };
};
}

void f(Outer::Inner x);
```

But I figured we could be more conservative and see if we actually have a reason to permit this as the current solution should already be sufficient for consuming C headers in C++ mode.

Reply:
Yeah, makes sense to be more conservative for now.